tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -598,3 +608,93 @@ def test_apply_delayed_percent_condition_comparison_interval(self):
         assert len(rule_fire_histories) == 1
         assert (percent_comparison_rule.id, group5.id) in rule_fire_histories
         self.assert_buffer_cleared(project_id=self.project.id)
+
+    def test_apply_delayed_no_bleed_count_and_comparison_condition(self):
+        """
+        Test that having both count and percent comparison type conditions do
+        not affect each other and the order the different types of conditions
+        are processed does not matter.
+
+        Also ensure the count query and first percent query share the same
+        query, but the second percent query is separate.
+        """
+
+        def ordered_callthrough(descending=False):
```

Comment:
🤔 can we call this `mock_get_condition_group`? or toss `side_effect`/`callback` somewhere in teh name of this?

Reply:
Also - does this need to be a separate helper?

Why are we modifying the args being passed into this method? This means we _aren't_ properly capturing how the implementation would behave.

This feels smelly 🤔

Reply:
We're modifying the order of the list of rules being passed into the function. We need to do this to test the behavior when count and percent conditions can share a query, so that ordering doesn't matter.

Originally they come in a random order, so we test for both orderings (count first then percent and vice versa)